Add log file for sql queries - [#5445] for 340

alikon opened the pull request:
Allow the Debug system plugin to log executed queries to a file
#5445 revisited for 340

alikon committed:
log sql for 340

brianteeman commented on administrator/language/en-GB/en-GB.plg_system_debug.ini:
```diff
@@ -9,6 +9,8 @@ PLG_DEBUG_ERRORS="Errors"
 PLG_DEBUG_EXPLAIN="Explain"
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_DESC="Optionally restrict users that can see debug information to those in the selected user groups. If none selected, all users will see the debug information."
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_LABEL="Allowed Groups"
+PLG_DEBUG_FIELD_EXECUTEDSQL_DESC="If enabled, executed SQL query will be logged. Only use this setting for short periods of time and for benchmarking purposes."
```
shouldnt this be SQL queries

brianteeman commented on administrator/language/en-GB/en-GB.plg_system_debug.ini:
```diff
@@ -9,6 +9,8 @@ PLG_DEBUG_ERRORS="Errors"
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_DESC="Optionally restrict users that can see debug information to those in the selected user groups. If none selected, all users will see the debug information."
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_LABEL="Allowed Groups"
+PLG_DEBUG_FIELD_EXECUTEDSQL_DESC="If enabled, executed SQL query will be logged. Only use this setting for short periods of time and for benchmarking purposes."
+PLG_DEBUG_FIELD_EXECUTEDSQL_LABEL="Log executed SQL"
```
Log Executed Queries

alikon committed:
English language correction

Corrected wrong english

zero-24 committed:
Update debug.php

zero-24 commented:
@alikon looks cool i have just a a smal CS PR against your branche :smile: https://github.com/alikon/joomla-cms/pull/3

alikon committed:
Merge pull request #3 from zero-24/patch-8

codestyle fix

Erftralle commented:
@test: Works fine for me. :+1:

pe7er commented:
@test
Patch works ok! 
New option  Plugin Manager: System - Debug "Log Executed Queries" set to YES
+ "Debug System" in Global Configuration set to ON 
= logfile generated with Sitename "Joomla 3.4 test" in file: /logs/Joomla 3.4 test_com_config.sql

Note: the filename contains (IMHO unwanted) spaces. 
Is it necesarry to include the site name in the file name?

pe7er committed:
Prevent spaces in file name

This patch replaces spaces in site name with underscores so that the file name does not contain underscores.

pe7er commented:
I've submitted a PR to your repo to prevent spaces in the file names: https://github.com/alikon/joomla-cms/pull/4

brianteeman commented:
Thanks for testing - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6202">issues.joomla.org/joomla-cms/6202</a>.</sub>

alikon committed:
Merge pull request #4 from pe7er/patch-10

Prevent spaces in file name

alikon commented:
@pe7er 
sitename was an easy way to let me quickly understand from where that log come from

pe7er commented:
@alikon, Yeah, I understand now. Thanks!

msdevcon commented:
@Test 
Patch works fine.
Including the "Sitename" in the Filename may not be the best solution, regard Special caracters etc.
